src/api.jl

```diff
@@ -3477,8 +3484,10 @@ The following example retrieves the upper bounds for the columns of the current
 XPRSchgbounds,
 XPRSgetlb.
 """
-function getub(prob::XpressProblem, _dbdu, first, last)
+function getub(prob::XpressProblem, first::Int, last::Int)
+    _dbdu = Vector{Float64}(undef, last - first + 1)
```

Comment:
For functions that query solution I find it very useful to have in-place versions.
so that I keep a single allocated vector and just update it in-place, its very useful if you are solving many-many-many LP's in sequence.
For those function I would vote for two implementations:
```julia
function getub(prob::XpressProblem, _dbdu::Vector{Float64}, first::Int, last::Int)
    @assert length(_dbdu) = last - first + 1
    @checked Lib.XPRSgetub(prob, _dbdu, first, last)
    return nothing
end
function getub(prob::XpressProblem, first::Int, last::Int)
    _dbdu = Vector{Float64}(undef, last - first + 1)
    getub(prob, _dbdu, first, last)
    return _dbdu
end
```
It is not essencial for getub, but for getting primals and duals its a must.